### DISCUSSION TITLE
merge?

### DISCUSSION THREAD

**Discussion by cubiq** [laksjdjf/IPAdapter-ComfyUI#34] [General]
Hey, I was in contact with comfyanonymous (author of comfyUI) and he seemed interested in integrating the IPAdapter directly into the core comfyui if someone rewrote it in a more comfyui friendly way. I started from your code and rewrote it to be more memory efficient and to not rely only on "cuda". I also asked comfyanonymous to do some changes to the clip vision so it would be easier to work with ipadapters. Lastly I added a "noise" option that basically sends a noisy uncond_image instead of an empty one (the results are pretty nice)... and of course English documentation :smile:  https://github.com/cubiq/ComfyUI_IPAdapter_plus

Anyway I believe comfyanonymous changed his mind and decided to keep the extension outside of comfyui and now we have two extensions for the same things. I was wondering if you are interested in discussing a possible merge of the two projects we could open a new repo where both of us can write to or something.

Thanks,
Matt

**Comment by laksjdjf** [comment]
Hi.
I've seen the changes and works you've made, and I think they're wonderful.

I'll think about what to do regarding this.

**Comment by cubiq** [comment]
thanks for considering it, don't feel pressured in any way. it was just an idea we can keep the projects separates if you prefer. Thanks for your great work anyway!

**Comment by laksjdjf** [comment]
hi, @cubiq.
Since your repository already covers the functions of my repository, how about I stop updating this one and direct people to your repository?

**Reply by cubiq** [reply]
you did an incredible work with the IPAdapter and I'd hate to see you go. I think there are still a few things that need fixing, if you close your repo would you be willing to collaborate on mine?

**Reply by laksjdjf** [reply]
Sure!
I just want to avoid double implementation when adding some new feature.

**Reply by cubiq** [reply]
let's do it then! I'm asking you to merge and work together since forever :smile: 

I started working on IPAdapter because comfyanonymous asked for a rewrite to be integrated into the main repo... but then he changed his mind.
